python-backend/app/api/mines/permits/permit_amendment/models/permit_amendment.py

```diff
@@ -82,42 +89,58 @@ def validate_type_code(self, key, permit_amendment_type_code):
-            if original_permit_amendment and original_permit_amendment.issue_date:
-                if original_permit_amendment.issue_date > issue_date.date():
-                    raise AssertionError(
-                        'Permit amendment issue date cannot be before the permits First Issued date.'
-                    )
-        if issue_date:
-            if issue_date.isoformat() == '9999-12-31':
-                raise AssertionError(
-                    'Permit amendment issue date should be set to null if not known.')
-            if issue_date > datetime.today():
-                raise AssertionError('Permit amendment issue date cannot be set to the future.')
-        return issue_date
-
-    @validates('authorization_end_date')
-    def validate_authorization_end_date(self, key, authorization_end_date):
-        if authorization_end_date and authorization_end_date.isoformat() == '9999-12-31':
-            raise AssertionError('Permit amendment end date should be set to null if not known.')
-        return authorization_end_date
+    # @validates('received_date')
+    # def validate_received_date(self, key, received_date):
+    #     if received_date:
+    #         if received_date.isoformat() == '9999-12-31':
+    #             raise AssertionError(
+    #                 'Permit amendment received date should be set to null if not known.')
+    #         if received_date > datetime.today():
+    #             raise AssertionError('Permit amendment received date cannot be set to the future.')
+    #     return received_date
+
+    # @validates('issue_date')
+    # def validate_issue_date(self, key, issue_date):
+    #     if self.permit_amendment_type_code != 'OGP':
+    #         original_permit_amendment = self.query.filter_by(permit_id=self.permit_id).filter_by(
+    #             permit_amendment_type_code='OGP').first()
+    #         if original_permit_amendment and original_permit_amendment.issue_date:
+    #             if original_permit_amendment.issue_date > issue_date.date():
+    #                 raise AssertionError(
+    #                     'Permit amendment issue date cannot be before the permits First Issued date.'
+    #                 )
+    #     if issue_date:
+    #         if issue_date.isoformat() == '9999-12-31':
+    #             raise AssertionError(
+    #                 'Permit amendment issue date should be set to null if not known.')
+    #         if issue_date > datetime.today():
+    #             raise AssertionError('Permit amendment issue date cannot be set to the future.')
+    #     return issue_date
+
+    # @validates('authorization_end_date')
+    # def validate_authorization_end_date(self, key, authorization_end_date):
+    #     if authorization_end_date and authorization_end_date.isoformat() == '9999-12-31':
+    #         raise AssertionError('Permit amendment end date should be set to null if not known.')
+    #     return authorization_end_date
```

Comment:
MarshmallowSchema.load() does not convert string to dates, until these validators fail, hopefully there is some config to help that along in the future.